src/coreclr/vm/dacenumerablehash.inl

```diff
@@ -165,53 +169,74 @@ void DacEnumerableHashTable<DAC_ENUM_HASH_ARGS>::GrowTable()
     // error to our caller.
     FAULT_NOT_FATAL();
 
+    auto curBuckets = GetBuckets();
+    DWORD cBuckets = (DWORD)((size_t*)curBuckets)[0];
+
     // Make the new bucket table larger by the scale factor requested by the subclass (but also prime).
-    DWORD cNewBuckets = NextLargestPrime(m_cBuckets * SCALE_FACTOR);
-    S_SIZE_T cbNewBuckets = S_SIZE_T(cNewBuckets) * S_SIZE_T(sizeof(PTR_VolatileEntry));
+    DWORD cNewBuckets = NextLargestPrime(cBuckets * SCALE_FACTOR);
+    // two extra slots - slot [0] contains the length of the table,
+    //                   slot [1] will contain the next version of the table if it resizes
+    S_SIZE_T cbNewBuckets = S_SIZE_T(cNewBuckets + 2) * S_SIZE_T(sizeof(PTR_VolatileEntry));
+
+    // REVIEW: I need a temp array here, relatively small (under 1K elements typically), is this the right heap for that?
```

Comment:
You should use regular C/C++ heap. The loader heaps are not designed for allocating temporary arrays like this.

Reply:
Are the tails long enough to make it worth it? Would it be better to just walk them as necessary?

Reply:
I was thinking about the old table, which would be at 2x load factor at this point and that we are holding up somebody's progress (most likely JIT), so should use a tail array as a guarantee against degenerate cases.

However, these chains would be in the new table, and load factor while resizing will be 0 - 0.5, so buckets should be short. There is always a possibility of a case where a lot of items form a chain, but after tweaking the hash function in the previous change, it will be statistically rare .
Also, doubling-up resize can really happen only a few times in the life of the module, so I think the risk of causing an observable pause is low.

I will switch this to just walking.

Reply:
In `System.Linq.Expressions.Tests`, which liberally uses generics, the max walk during individual resizes is generally 1 - 5, but every run has some outliers with max walk > 50. I think it is acceptable. 

It makes me wonder though if there is a cheap way to bring the clustering further down by improving the hash.

Reply:
Hmm, the outliers are all from a different hashtable - `EEClassHashTable`, which shares the base implementation with `EETypeHashTable`, but has different elements and hash function.

Its resize factor is 4x, triggered at the same 2x load factor. In theory it should have even shorter buckets at resize. 
I will have to look closer.

Reply:
We are hashing types by namespace/name and see a lot of types with empty namespace and names like  "<>c" .

Reply:
`EEClassHashTable` seems to be intentionally placing same-named nested classes in the same bucket. 
I think walking the chain is still acceptable. - If these chains are acceptable for lookups, then they are ok for very rare 4x resizes.  

`EEClassHashTable` could use some clean-up in a separate change.

Reply:
I have a change that fixes the nested type collision issue for `EEClassHashTable`. It will be a separate PR.